packages/components/src/form-token-field/types.ts

```diff
@@ -154,9 +158,15 @@ export interface FormTokenFieldProps
 	 * @default false
 	 */
 	__next36pxDefaultSize?: boolean;
+	/**
+	 * Custom renderer for suggestions.
+	 */
+	__experimentalRenderItem?: ( args: { item: string } ) => ReactNode;
 }
 
-export interface SuggestionsListProps< T = string | { value: string } > {
+export interface SuggestionsListProps<
+	T = string | ( Record< string, unknown > & { value: string } )
```

Comment:
Maybe we could a comment here about the fact that `T` can be either a `string` or `Record< string, unknown > & { value: string }` because of its usage in both `FormTokenField` (as a `string`) and `ComboboxControl` (as an object with the `value` prop) (or at least, this is what I'm gathering from the code)